internal/service/grafana/workspace_api_key_test.go

```diff
@@ -0,0 +1,70 @@
+package grafana_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/managedgrafana"
+	sdkacctest "github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+)
+
+func testAccWorkspaceAPIKey_basic(t *testing.T) {
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	resourceName := "aws_grafana_workspace_api_key.test"
+	workspaceResourceName := "aws_grafana_workspace.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { acctest.PreCheck(t); acctest.PreCheckPartitionHasService(managedgrafana.EndpointsID, t) },
+		ErrorCheck:        acctest.ErrorCheck(t, managedgrafana.EndpointsID),
+		CheckDestroy:      nil,
+		ProviderFactories: acctest.ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccWorkspaceAPIKeyConfig_basic(rName, "test-api-1"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "key_name", "test-api-1"),
+					resource.TestCheckResourceAttr(resourceName, "key_role", "EDITIR"),
```

Comment:
This can be changed to use the value that the SDK provides

```
const (
    // RoleAdmin is a Role enum value
    RoleAdmin = "ADMIN"

    // RoleEditor is a Role enum value
    RoleEditor = "EDITOR"

    // RoleViewer is a Role enum value
    RoleViewer = "VIEWER"
)
```

Reply:
Done as requested.